internal/cli/cmd/cluster/buildproxy.go

```diff
@@ -118,32 +130,80 @@ func NewBuildClusterInstance0(p api.BuildPlatform) *BuildClusterInstance {
 }
 
 type buildProxy struct {
-	socketPath       string
-	instance         *BuildClusterInstance
-	listener         net.Listener
-	cleanup          func() error
-	useGrpcProxy     bool
-	injectWorkerInfo *controlapi.ListWorkersResponse
+	socketPath        string
+	controlSocketPath string
+	instance          *BuildClusterInstance
+	listener          net.Listener
+	cleanup           func() error
+	useGrpcProxy      bool
+	injectWorkerInfo  *controlapi.ListWorkersResponse
+	proxyStatus       *proxyStatusDesc
+}
+
+// proxyStatus is used by `nsc docker buildx status` to show user info on
+// proxy current status
+type proxyStatusDesc struct {
+	mu                *sync.RWMutex
```

Comment:
Nit: make this `mu sync.RWMutex`; it will be a pointer because `proxyStatusDesc` is always used by reference.